ophyd/status.py

```diff
@@ -97,15 +117,40 @@ def done(self, value):
             "to fail.",
             UserWarning)
 
+    @property
+    def success(self):
+        """
+        Boolean indicating whether associated operation has completed.
+
+        The method :meth:`exception` provides more specific information about
+        failure.
+
+        Once it is set to True, it can never change to be False.
+        """
+        return self.done and self._exception is None
+
+    @success.setter
+    def success(self, value):
+        # For now, allow this setter to work only if it has no effect.
+        # In a future release, make this property not settable.
+        if bool(self.success) != bool(value):
+            raise RuntimeError(
+                "The success state of a status object cannot be changed by "
+                "setting its `success` attribute directly. Call `_finished()`.")
+        warn(
+            "Do not set the `succcess` attribute of a status object directly. "
+            "It should only be set indirectly by calling `_finished()`. "
```

Comment:
This warning and the above error should refer to the newly added `set_finished` method instead of the older `_finished` method